Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2218,6 +2223,10 @@ - (void)collectionView:(JSQMessagesCollectionView *)collectionView
     // while updating the range and the dynamic interactions.
     [[NSNotificationCenter defaultCenter] removeObserver:self name:YapDatabaseModifiedNotification object:nil];
 
+    // We need to `beginLongLivedReadTransaction` before we update our
+    // mapping in order to jump to the most recent commit.
+    [self.uiDatabaseConnection beginLongLivedReadTransaction];
```

Comment:
Same as above - `ensureDynamicInteractions` should operate on the latest thread state.